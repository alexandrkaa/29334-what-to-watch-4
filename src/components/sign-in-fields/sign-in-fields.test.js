import React from 'react';
import renderer from 'react-test-renderer';
import SignInFields from './sign-in-fields.jsx';

describe(`<SignInFields /> should match snapshot`, () => {
  const onInputChange = jest.fn();
  const FieldsIds = {
    EMAIL_FIELD_ID: `user-email`,
    PASSWORD_FIELD_ID: `user-password`,
  };

  const fields = [
    {
      id: FieldsIds.EMAIL_FIELD_ID,
      label: `Email address`,
      type: `email`,
      placeholder: `Email address`,
      value: ``,
    },
    {
      id: FieldsIds.PASSWORD_FIELD_ID,
      label: `Password`,
      type: `password`,
      placeholder: `Password`,
      value: ``,
    },
  ];
  it(`<AddCommentForm /> should match snapshot not disabled`, () => {
    const isLoading = false;
    const tree = renderer
      .create(
          <SignInFields
            signInFields={fields}
            isLoading={isLoading}
            onInputChange={onInputChange}
          />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it(`<AddCommentForm /> should match snapshot not disabled`, () => {
    const isLoading = true;
    const tree = renderer
      .create(
          <SignInFields
            signInFields={fields}
            isLoading={isLoading}
            onInputChange={onInputChange}
          />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
