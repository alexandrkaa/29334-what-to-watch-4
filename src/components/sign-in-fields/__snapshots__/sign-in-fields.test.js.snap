// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignInFields /> should match snapshot <AddCommentForm /> should match snapshot not disabled 1`] = `
<div
  className="sign-in__fields"
>
  <div
    className="sign-in__field"
  >
    <input
      className="sign-in__input"
      disabled={false}
      id="user-email"
      name="user-email"
      onChange={[Function]}
      placeholder="Email address"
      type="email"
      value=""
    />
    <label
      className="sign-in__label visually-hidden"
      htmlFor="user-email"
    >
      Email address
    </label>
  </div>
  <div
    className="sign-in__field"
  >
    <input
      className="sign-in__input"
      disabled={false}
      id="user-password"
      name="user-password"
      onChange={[Function]}
      placeholder="Password"
      type="password"
      value=""
    />
    <label
      className="sign-in__label visually-hidden"
      htmlFor="user-password"
    >
      Password
    </label>
  </div>
</div>
`;

exports[`<SignInFields /> should match snapshot <AddCommentForm /> should match snapshot not disabled 2`] = `
<div
  className="sign-in__fields"
>
  <div
    className="sign-in__field"
  >
    <input
      className="sign-in__input"
      disabled={true}
      id="user-email"
      name="user-email"
      onChange={[Function]}
      placeholder="Email address"
      type="email"
      value=""
    />
    <label
      className="sign-in__label visually-hidden"
      htmlFor="user-email"
    >
      Email address
    </label>
  </div>
  <div
    className="sign-in__field"
  >
    <input
      className="sign-in__input"
      disabled={true}
      id="user-password"
      name="user-password"
      onChange={[Function]}
      placeholder="Password"
      type="password"
      value=""
    />
    <label
      className="sign-in__label visually-hidden"
      htmlFor="user-password"
    >
      Password
    </label>
  </div>
</div>
`;
